main.go

```diff
@@ -187,86 +183,96 @@ func main() {
 		os.Exit(1)
 	}
 
-	sysPromptInput := fmt.Sprintf(`{"message":%q,"fields":""}`, fmt.Sprintf("Opening browser to %s. If there is an issue, paste this link into a browser manually.", u.String()))
+	metadata := map[string]string{
+		"toolContext":     "credential",
+		"toolDisplayName": fmt.Sprintf("%s%s Integration", strings.ToTitle(integration[:1]), integration[1:]),
+		"authURL":         u.String(),
+	}
+
+	b, err := json.Marshal(metadata)
+	if err != nil {
+		fmt.Printf("failed to marshal metadata: %v\n", err)
+		os.Exit(1)
+	}
 
 	run, err := gs.Run(context.Background(), "sys.prompt", gptscript.Options{
-		Input: sysPromptInput,
+		Input: fmt.Sprintf(`{"metadata":%s,"message":%q}`, b, fmt.Sprintf("Opening browser to %s. If there is an issue, paste this link into a browser manually.", u.String())),
 	})
 	if err != nil {
 		fmt.Printf("failed to run sys.prompt: %v\n", err)
 		os.Exit(1)
 	}
 
-	if _, err = run.Text(); err != nil {
+	out, err := run.Text()
+	if out != "" && err != nil {
```

Comment:
Why do we only care about the error if the output is non-empty? I feel like we should be checking the error every time here.

Reply:
Yep. Good call.